classes/submission/maps/Schema.php

```diff
@@ -103,4 +126,100 @@ protected function getReviewRoundsFromSubmission(Submission $submission): Collec
         return collect($reviewRoundDao->getBySubmissionId($submission->getId())->toIterator())
             ->groupBy(fn (ReviewRound $reviewRound) => $reviewRound->getData('round'));
     }
+
+    /**
+     * Gets the Editorial decisions available to editors for a given stage of a submission
+     *
+     * This method returns decisions only for active stages. For inactive stages, it returns an empty array.
+     *
+     * @return DecisionType[]
+     *
+     * @hook Workflow::Decisions [[&$decisionTypes, $stageId]]
+     */
+    protected function getAvailableEditorialDecisions(int $stageId, Submission $submission): array
+    {
+        $request = Application::get()->getRequest();
+        $user = $request->getUser();
+        $isActiveStage = $submission->getData('stageId') == $stageId;
+        $permissions = $this->checkDecisionPermissions($stageId, $submission, $user, $request->getContext()->getId());
+        $userHasAccessibleRoles = $user->hasRole([Role::ROLE_ID_SUB_EDITOR, Role::ROLE_ID_MANAGER, Role::ROLE_ID_SITE_ADMIN, Role::ROLE_ID_ASSISTANT], $request->getContext()->getId());
+
+        if (!$userHasAccessibleRoles || !$isActiveStage || !$permissions['canMakeDecision']) {
+            return [];
+        }
+
+        $decisionTypes = []; /** @var DecisionType[] $decisionTypes */
+
+        $reviewRoundDao = DAORegistry::getDAO('ReviewRoundDAO'); /** @var ReviewRoundDAO $reviewRoundDao */
+        $reviewRound = $reviewRoundDao->getLastReviewRoundBySubmissionId($submission->getId(), $stageId);
+
+        $isOnlyRecommending = $permissions['isOnlyRecommending'];
+
+        if ($isOnlyRecommending && $stageId === WORKFLOW_STAGE_ID_SUBMISSION) {
```

Comment:
For reference, see https://github.com/pkp/pkp-lib/blob/9e75633e4809dc79ce26b8e8a3287073478020bb/pages/workflow/PKPWorkflowHandler.php#L599-L607